### ISSUE TITLE
Failed to compile when installing METAVIRALSPADES

### ISSUE THREAD

**Issue by ZhangDengwei** [ablab/spades#639]
Hi, 

When compiling the `metaviralspades` with the command of `./spades_compile.sh`, I got the following errors

```
lkit/spades/spades-metaviral_publication/assembler/src"
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Setting default build configuration: RelWithAsserts
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11")
-- Found BZip2: /usr/lib/x86_64-linux-gnu/libbz2.so (found version "1.0.6")
-- Looking for BZ2_bzCompressInit
-- Looking for BZ2_bzCompressInit - found
-- Found readline: /usr/include
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libncurses.so
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libncurses.so - found
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libcurses.so
-- Looking for wsyncup in /usr/lib/x86_64-linux-gnu/libcurses.so - found
-- Found Boost 1.70.0 at /home/dengwei/toolkit/anaconda/lib/cmake/Boost-1.70.0
--   Requested configuration: QUIET REQUIRED
-- Found boost_headers 1.70.0 at /home/dengwei/toolkit/anaconda/lib/cmake/boost_headers-1.70.0
-- Boost  found.
Making Release Configuration...
CMake Error at cmake/includes.cmake:8 (include_directories):
  include_directories given empty-string as include directory.
Call Stack (most recent call first):
  CMakeLists.txt:58 (include)


Building SPAdes from N/A, sha: N/A
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of void *
-- Check size of void * - done
-- Performing Test HAVE_CPU_SPINWAIT
-- Performing Test HAVE_CPU_SPINWAIT - Success
-- System addr bits: 48
-- Page size: 4096 bytes
-- Cache Line size: 64 bytes
-- Performing Test JEMALLOC_USE_SYSCALL
-- Performing Test JEMALLOC_USE_SYSCALL - Success
-- Looking for log in m
-- Looking for log in m - found
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - found
-- Performing Test malloc_usable_size_const
-- Performing Test malloc_usable_size_const - Failed
-- Check if the system is big endian
-- Searching 16 bit integer
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Performing Test JEMALLOC_HAVE_ATTR
-- Performing Test JEMALLOC_HAVE_ATTR - Success
-- Performing Test HAVE_TLS_MODEL
-- Performing Test HAVE_TLS_MODEL - Success
-- Performing Test JEMALLOC_HAVE_ATTR_ALLOC_SIZE
-- Performing Test JEMALLOC_HAVE_ATTR_ALLOC_SIZE - Success
-- Performing Test JEMALLOC_HAVE_ATTR_FORMAT_GNU_PRINTF
-- Performing Test JEMALLOC_HAVE_ATTR_FORMAT_GNU_PRINTF - Success
-- Performing Test JEMALLOC_HAVE_ATTR_FORMAT_PRINTF
-- Performing Test JEMALLOC_HAVE_ATTR_FORMAT_PRINTF - Success
-- Performing Test JEMALLOC_HAS_RESTRICT
-- Performing Test JEMALLOC_HAS_RESTRICT - Success
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Performing Test JEMALLOC_HAVE_CLOCK_MONOTONIC
-- Performing Test JEMALLOC_HAVE_CLOCK_MONOTONIC - Success
-- Performing Test JEMALLOC_HAVE_CLOCK_MONOTONIC_COARSE
-- Performing Test JEMALLOC_HAVE_CLOCK_MONOTONIC_COARSE - Success
-- Performing Test JEMALLOC_HAVE_MACH_ABSOLUTE_TIME
-- Performing Test JEMALLOC_HAVE_MACH_ABSOLUTE_TIME - Failed
-- Performing Test HAS_BUILTIN_FFSL
-- Performing Test HAS_BUILTIN_FFSL - Success
-- Performing Test JEMALLOC_INTERNAL_POPCOUNT
-- Performing Test JEMALLOC_INTERNAL_POPCOUNT - Success
-- Performing Test JEMALLOC_INTERNAL_POPCOUNTL
-- Performing Test JEMALLOC_INTERNAL_POPCOUNTL - Success
-- Looking for memalign
-- Looking for memalign - not found
-- Looking for valloc
-- Looking for valloc - found
-- Looking for secure_getenv
-- Looking for secure_getenv - found
-- Looking for sched_getcpu
-- Looking for sched_getcpu - found
-- Looking for sched_setaffinity
-- Looking for sched_setaffinity - found
-- Looking for issetugid
-- Looking for issetugid - not found
-- Looking for _malloc_thread_cleanup
-- Looking for _malloc_thread_cleanup - not found
-- Looking for _pthread_mutex_init_calloc_cb
-- Looking for _pthread_mutex_init_calloc_cb - not found
-- Performing Test HAS_MADVISE_0
-- Performing Test HAS_MADVISE_0 - Success
-- Performing Test HAS_MADVISE_MADV_FREE
-- Performing Test HAS_MADVISE_MADV_FREE - Success
-- Performing Test HAS_MADVISE_MADV_DONTNEED
-- Performing Test HAS_MADVISE_MADV_DONTNEED - Success
-- Performing Test HAS_MADVISE_MADV_DONTDUMP
-- Performing Test HAS_MADVISE_MADV_DONTDUMP - Success
-- Performing Test HAS_MADVISE_MADV_DODUMP
-- Performing Test HAS_MADVISE_MADV_DODUMP - Success
-- Performing Test HAS_MADVISE_MADV_HUGEPAGE
-- Performing Test HAS_MADVISE_MADV_HUGEPAGE - Success
-- Performing Test HAS_MADVISE_MADV_NOHUGEPAGE
-- Performing Test HAS_MADVISE_MADV_NOHUGEPAGE - Success
-- Performing Test JEMALLOC_HAVE_BUILTIN_CLZ
-- Performing Test JEMALLOC_HAVE_BUILTIN_CLZ - Success
-- Performing Test JEMALLOC_OS_UNFAIR_LOCK
-- Performing Test JEMALLOC_OS_UNFAIR_LOCK - Failed
-- Performing Test JEMALLOC_GLIBC_MALLOC_HOOK
-- Performing Test JEMALLOC_GLIBC_MALLOC_HOOK - Success
-- Performing Test has_memalign_hook
-- Performing Test has_memalign_hook - Success
-- Performing Test JEMALLOC_HAVE_PTHREAD_ATFORK
-- Performing Test JEMALLOC_HAVE_PTHREAD_ATFORK - Success
-- Performing Test JEMALLOC_HAVE_PTHREAD_SETNAME_NP
-- Performing Test JEMALLOC_HAVE_PTHREAD_SETNAME_NP - Success
-- Performing Test JEMALLOC_HAVE_PTHREAD_MUTEX_ADAPTIVE_NP
-- Performing Test JEMALLOC_HAVE_PTHREAD_MUTEX_ADAPTIVE_NP - Success
-- Performing Test JEMALLOC_STRERROR_R_RETURNS_CHAR_WITH_GNU_SOURCE
-- Performing Test JEMALLOC_STRERROR_R_RETURNS_CHAR_WITH_GNU_SOURCE - Success
-- Performing Test JEMALLOC_OSATOMIC
-- Performing Test JEMALLOC_OSATOMIC - Failed
-- Performing Test JEMALLOC_GCC_ATOMIC_ATOMICS
-- Performing Test JEMALLOC_GCC_ATOMIC_ATOMICS - Success
-- Performing Test JEMALLOC_TLS
-- Performing Test JEMALLOC_TLS - Success
-- Performing Test JEMALLOC_GCC_U8_ATOMIC_ATOMICS
-- Performing Test JEMALLOC_GCC_U8_ATOMIC_ATOMICS - Success
-- Performing Test JEMALLOC_GCC_SYNC_ATOMICS
-- Performing Test JEMALLOC_GCC_SYNC_ATOMICS - Success
-- Performing Test JEMALLOC_GCC_U8_SYNC_ATOMICS
-- Performing Test JEMALLOC_GCC_U8_SYNC_ATOMICS - Success
-- Performing Test has__builtin_unreachable
-- Performing Test has__builtin_unreachable - Success
-- Performing Test JEMALLOC_C11_ATOMICS
-- Performing Test JEMALLOC_C11_ATOMICS - Success
-- Check size of void *
-- Check size of void * - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of long long
-- Check size of long long - done
-- Check size of intmax_t
-- Check size of intmax_t - done
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for dlsym
-- Looking for dlsym - found
-- Performing Test _std_gnu11
-- Performing Test _std_gnu11 - Success
-- Performing Test _Wall
-- Performing Test _Wall - Success
-- Performing Test _Wextra
-- Performing Test _Wextra - Success
-- Performing Test _Wshorten_64_to_32
-- Performing Test _Wshorten_64_to_32 - Failed
-- Performing Test _Wsign_compare
-- Performing Test _Wsign_compare - Success
-- Performing Test _Wundef
-- Performing Test _Wundef - Success
-- Performing Test _Wno_format_zero_length
-- Performing Test _Wno_format_zero_length - Success
-- Performing Test _pipe
-- Performing Test _pipe - Success
-- Performing Test _g3
-- Performing Test _g3 - Success
-- Performing Test _O3
-- Performing Test _O3 - Success
-- Performing Test _funroll_loops
-- Performing Test _funroll_loops - Success
-- Performing Test HAS_CXX14
-- Performing Test HAS_CXX14 - Success
-- Found Git: /usr/bin/git (found version "2.17.1")
-- Looking for memalign
-- Looking for memalign - found
-- Looking for valloc
-- Looking for valloc - found
-- Translating jemalloc_macros.h.in -> jemalloc_macros.h.in.cmake => jemalloc_macros.h
-- Translating jemalloc_defs.h.in -> jemalloc_defs.h.in.cmake => jemalloc_defs.h
-- Translating jemalloc_protos.h.in -> jemalloc_protos.h.in.cmake => jemalloc_protos.h
-- Translating jemalloc_typedefs.h.in -> jemalloc_typedefs.h.in.cmake => jemalloc_typedefs.h
-- Translating jemalloc_preamble.h.in -> jemalloc_preamble.h.in.cmake => jemalloc_preamble.h
-- Translating jemalloc_internal_defs.h.in -> jemalloc_internal_defs.h.in.cmake => jemalloc_internal_defs.h
-- Translating jemalloc_test_defs.h.in -> jemalloc_test_defs.h.in.cmake => jemalloc_test_defs.h
-- Configuring jemalloc_test.h.in -> jemalloc_test.h
-- Translating jemalloc_protos.h.in -> jemalloc_protos.h.in.cmake => jemalloc_protos_jet.h
-- Looking for open
-- Looking for open - found
-- Looking for include file execinfo.h
-- Looking for include file execinfo.h - found
-- Looking for include file signal.h
-- Looking for include file signal.h - found
-- Looking for include file sys/stat.h
-- Looking for include file sys/stat.h - found
-- Looking for include file cxxabi.h
-- Looking for include file cxxabi.h - not found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file mach/mach.h
-- Looking for include file mach/mach.h - not found
-- Looking for include file link.h
-- Looking for include file link.h - found
-- Looking for backtrace
-- Looking for backtrace - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Looking for shm_open
-- Looking for shm_open - not found
-- Looking for shm_open in rt
-- Looking for shm_open in rt - found
-- Configuring incomplete, errors occurred!
```

I guess the error occurred because of the `cmake` whose version is 3.13.2. Could you give me some suggestions to help me out? Alternatively, may I just copy the script `metaplasmidspades.py` into another SPADES folder that was previously installed? 

Chress!

**Comment by asl** [comment]
Tagging @Dmitry-Antipov for help

**Comment by Dmitry-Antipov** [comment]
Hi.
Can you please try recently released SPAdes 3.15 (https://cab.spbu.ru/software/spades/) ? MetaplasmidSPAdes is included.

Yes, for me it looks like cmake problem and copying of this script definitely will not help. But I hope that 3.15 should be OK.

**Comment by ZhangDengwei** [comment]
It works for me. Many thanks!!!
